### DISCUSSION TITLE
Recommended USB Grabber for HDR/Dolby Vision

### DISCUSSION THREAD

**Discussion by rlg6767** [awawa-dev/HyperHDR#1016] [General]
Hi.    First off what a great project and massive thanks to the developers for this product!

I have been ploughing through a lot of useful guides, discussion forums & articles but I am still a bit stumped when it comes to which USB grabber is best for my needs.  Examples of the guides I am referring to are [The Holy Grail of HDR: support for P010 format](https://github.com/awawa-dev/HyperHDR/discussions/967) and [MS2130 grabber. Summary and firmware testing](https://github.com/awawa-dev/HyperHDR/discussions/729) to name a couple but I am still a bit lost.

I have been running HyperHDR for some time using an EZCOO splitter (EZ-SP12H2) and I recently purchased a grabber (UGreen CM360) which works perfectly for SDR but doesn't work so well for HDR (washed out) and not at all for Dolby Vision.  I appreciate that I can callibrate LUT tables for the HDR side of things but my grabber flat out refuses to work with Dolby Vision   My requirement is for screen grabbing from my Nvidia Shield (4K 60Hz,  SDR,  HDR and Dolby Vision) and my Xbox One (4K 60Hz).   What is the best grabber for my needs please?   I can see a lot of talk about new products coming on the market and things seem to be moving quite quickly hence why I am little confused and could with some help.

For reference I am running HyperHDR 21 beta as a Proxmox container but could easily switch to a Raspberry Pi if that better suits my needs.

Many thanks!

**Comment by awawa-dev** [comment]
Hi
As for Dolby Vision (only LLDV is supported), what are your symptoms that it doesn't work? It's the HDMI splitter's job, not the grabber's job, to handle LLDV correctly, so replacing the grabber I don't know if it will change anything (although it can't be ruled out if the EDID grabber changes).

> doesn't work so well for HDR (washed out)

Have you actually enabled tone mapping? Can you pause the video player and take screenshots of the live preview in HyperHDR with and without tone mapping enabled? Have you calibrated the LUT for HDR?
Usually in the case of Ezcoo HDR and DV(LLDV) use the same LUT table, LLDV is basically HDR10 without meta tags.

I currently use UGREEN 25173 but only because of P010 format. If I had to rate it, it would look like this:
the old Elgato HD60X is much better than UGreen 25173 especially in modes other than P010, but UGreen 25173 is better or comparable to the new Elgato HD60X which lacks P010 scaling.

BTW, can you check the "lsusb" command to see what grabber IDs are there?

**Reply by rlg6767** [reply]
Hi,  many thanks for your reply.

> what are your symptoms that it doesn't work?

I know my splitter is working correctly as when I set my Nvidia Shield to 60Hz HDR mode I can see the live preview in HyperHDR.   As soon as I enable Dolby Vision on the Shield then I get a black screen with the message "Video not Supported" or something along those line.

> Have you actually enabled tone mapping?

Yes I have,  tried with it enabled and disabled.

> Can you pause the video player and take screenshots of the live preview in HyperHDR with and without tone mapping enabled? Have you calibrated the LUT for HDR?

I'm working at the minute but I should free in a few hours to post the results.

> Have you calibrated the LUT for HDR?

Not yet,   so I suspect I can probably fix HDR but the Dolby Vision piece is still an issue.   Will try and do this today.

> can you check the "lsusb" command

![image](https://github.com/user-attachments/assets/25d6eb4c-59a3-4ebb-84c6-ebc911d2d598)

**Reply by awawa-dev** [reply]
Oh, hello MS2131 :-) It's almost the same as MS2130, except it offers HDMI pass-through. I once did an experiment where I uploaded MS2130 firmware (our experimental F3) to it and the grabber worked like a regular MS2130, although of course the HDMI pass-through didn't work. After restoring the previously made MS2131 backup, its old functions returned. Of course I don't encourage such experiments ;-)

> I know my splitter is working correctly as when I set my Nvidia Shield to 60Hz HDR mode I can see the live preview in HyperHDR. As soon as I enable Dolby Vision on the Shield then I get a black screen with the message "Video not Supported" or something along those line.

Have you connected the grabber to Ezcoo output 2 and enabled the scaler? TV is connected to Ezcoo output 1 with scaler disabled and output of the UGreen grabber is not connected?

**Reply by rlg6767** [reply]
> Did you connected the grabber to Ezcoo output 2 and enabled the scaler? TV is connected to Ezcoo output 1 with scaler disabled and output of the UGreen grabber is not connected?

That's exactly what I have done,  grabber is just accepting an input and nothing connected to its output.   I am more than happy to experiment with firmware if you think it's worth a shot (and I promise I won't hold you accountable!! :-)).   The device cost me £20 from Ali Express and is a drop in the ocean compared to what I have spent on my smart home tech...

**Reply by awawa-dev** [reply]
I'm not sure why LLDV doesn't want to work for you. Maybe the grabber is acting strange if it doesn't have an output connected. But do you have scaler 2 on Ezcoo enabled?

**Reply by rlg6767** [reply]
Hey.    Will test again in a couple of hours but yes.  Ezcoo output 1 set to 4K and output 2 set to downscale to 1080p.   There's nothing else I need to do with the dip switches for EDID on the Ezcoo?  I read the manual but fooked if I could understand them.  They are set to default.

**Reply by rlg6767** [reply]
OK, Have made a breakthrough!!   I finished work a bit early and have just tested and found that the output of the UGreen does need to be connected to something for it to work.  I connected it to a spare port on my AVR and voila I now have screen capturing of Dolby Vision.   I will post some screenshots of HDR tone mapping enabled/disabled to see if it needs further tweaking!!

**Reply by AstaRom** [reply]
![Screenshot_2](https://github.com/user-attachments/assets/a53c8355-f02a-4a5c-a555-8dde9200e438)

**Reply by rlg6767** [reply]
OK,  so this is getting weird.    It was working fine for 1080p and HDR/DV content.   Turned the TV off and back on again 15 mins later and no backlight.   Checked HyperHDR and it's not detecting any video.   I went to restart the docker and got this:
![image](https://github.com/user-attachments/assets/efa4b1d1-5efe-4d2c-afca-8c98e87e0d65)

Running lsusb gives me this:
![image](https://github.com/user-attachments/assets/719579a8-4b94-4ab1-9228-6a9918b633bb)

Checking v4l2 gives me this:
![image](https://github.com/user-attachments/assets/0e6630d8-71eb-4a8f-9cb8-df93f93f7ad3)

Disconnecting and reconnecting grabber gives me the right info for v4l2 and allows me to restart HyperHDR but I still get this in live video:
![image](https://github.com/user-attachments/assets/24bc10ed-58ac-4735-ae72-c0dc3fcaeb35)

Don't get it..   Playing the same content that was working fine an hour ago...

**Reply by rlg6767** [reply]
OK rebooting the VM has fixed the problem.  I am wondering whether this could be a grabber firmware problem?   It seems to be reoccurring.

**Reply by awawa-dev** [reply]
Next time please check dmesg too. The grabber or its firmware could be a problem but it can be also insufficient power supply or bad USB cable. Or the VM pass-through itself.

**Reply by rlg6767** [reply]
Losing the will to live with this.   Grabber was working fine for a while and then I had to reboot my Proxmox server.   For some reason /dev/video0 or 1 doesn't get created although lsusb sees the device.  So I can't pass the device through to the lxc.

lsusb:
![image](https://github.com/user-attachments/assets/f5e7fac0-f6f8-433f-9b46-b6c8b4f8296c)

dmesg:
![image](https://github.com/user-attachments/assets/630a1da9-a4c1-4cd4-84dc-0b4539831dfd)

Have tried plugging/re-plugging and rebooting.   No other VM or LXC has this device in their device config.

**Reply by awawa-dev** [reply]
Did it reconnect in 413 seconds after it had previously disconnected (it's not in this part of dmesg, it was there before that)? I think that's the main problem: the disconnections. Are you using a usb hub?

**Reply by rlg6767** [reply]
That may have been a reboot I did?  I am not using a hub.    Let me do a disconnect and reconnect and repost the dmesg output.

**Reply by rlg6767** [reply]
So have just done a disconnect and reconnect approximately 10 seconds apart

![image](https://github.com/user-attachments/assets/7bae3413-27c0-4e19-ada0-0fcf4e02154a)

**Reply by awawa-dev** [reply]
Rather it's not a reboot because it's late, at @413 second. Resetting VM should not have a such effect but I'm not sure.

**Reply by awawa-dev** [reply]
Probably VM is not smart enough to re-pass-though the video device after it disappears and re-appears. But the main problem is that the grabber disconnects from USB by itself at some point. I encountered a similar problem with another user, but it was MS2130 and the problem was a damaged grabber and replacing it helped (but I would leave that for the very end)

**Reply by rlg6767** [reply]
So I just built a new VM and passed through the USB device.  I saw in DMESG on bootup that it threw this message:

`usb can't set config #1 error 32`

Unplugging and replugging it then got it working again and I can see live video in HyperHDR.

It's definitely temperamental though.   Do you think it is worth flashing the firmware on the grabber?

**Reply by awawa-dev** [reply]
Did it behave similarly if the grabber did not have an HDMI cable connected to the AVR (dummy connection)?

**Reply by rlg6767** [reply]
I just pulled the dummy connection and it's working fine.

**Reply by awawa-dev** [reply]
If after disconnecting the HDMI cable from the AVR the grabber will work correctly for a longer period of time (without any disconnections) then this may be the problem. Maybe the AVR resets unused HDMI after some time. And only then I would consider changing the firmware to get rid of the need to use the unused grabber HDMI pass-through by "downgrading" to MS2130.

**Reply by awawa-dev** [reply]
>  and I recently purchased a grabber (UGreen CM360) which works perfectly for SDR but doesn't work so well for HDR (washed out) and not at all for Dolby Vision.

You didn't connect HDMI at that time so it worked without disconnecting?

**Reply by rlg6767** [reply]
It was working before with it connected.  It is still working now with it disconnected.

**Reply by rlg6767** [reply]
And this is why I am suspecting firmware.  It's just so temperamental.

**Reply by awawa-dev** [reply]
If you are considering changing the firmware, from what I remember I used Windows flasher (from MS2130 thread), you had to change the product id to 2131 (PID in the connect section, also set MS213x in the top section) in options because the standard one is 2130 and it didn't see the device. Then backup old firmware and flash MS2130. After finishing & disconnecting/reconnecting device, the flasher doesn't see the device again because the product id changed from 2131 to 2130. I only did it for MS2131 Hagibis, not UGreen, I don't know if they changed something.

**Reply by rlg6767** [reply]
OK many thanks.   Will do this as a last resort!!

**Reply by rlg6767** [reply]
So can you give me a bit more detail on the backup and flash process please?   I have set the MS2131 in the top secion and changed the product id.   What step to backup?  And then how to reflash?

![image](https://github.com/user-attachments/assets/b2b43152-21b3-4bb5-8f12-cb5b64e0f34c)

**Reply by awawa-dev** [reply]
now click connect

**Reply by rlg6767** [reply]
Managed to connect and clicked read.  But the corresponding output was all zero's so maybe not compatible.

**Reply by awawa-dev** [reply]
Give me a minute. I'll find my ms2131

**Reply by awawa-dev** [reply]
Select 'Flash' option, not default 24c01, then click read again

**Reply by rlg6767** [reply]
OK will do that after dinner.    Thankyou for checking that!   Do you also select 'Flash' option when writing the new firmware?

**Reply by awawa-dev** [reply]
Yes. First `read` using mentioned settings (after `Connect`) must be successful, then `save to bin` to backup firmware, next `open file` f3 firmware bin file (make sure you extracted it from zip). And finally `Download` to flash the firmware. Once done, disconnect/reconnect the grabber. Change the product ID to 2130 and connect to verify it works.

**Reply by rlg6767** [reply]
Have one last thing, well two things, I want to try before I flash the grabber.  I have a small NUC that I have installed Ubuntu directly on,  will try to see if that helps to rule out Proxmox/Virtualisation etc.   I also tried my laptop running Win 11 and that seemed to work flawlessly,  albeit with fairly limited testing.  If bare metal Ubuntu doesn't work so well will install Windows on the NUC and test these out and report back.   If neither of these work,  will proceed with flashing.

**Reply by awawa-dev** [reply]
Yes, leave the flashing for the very end. And on what unit did you previously test the MS2131 and there were these errors: Raspberry Pi? If so, it is very possible that it is a power supply issue, MS213x needs much more power than its predecessor MS2109

**Reply by rlg6767** [reply]
It was Ubuntu 22.04 running as a VM in Proxmox previously.  I also tried Ubuntu 24.04 VM also on Proxmox.   Last night I fired up HyperHDR on my Win11 laptop and it just seemed to "work".   Will continue to test over the weekend and report back with results.

**Reply by rlg6767** [reply]
OK so I have succesfully flashed the device now.   Just doing the calibration for HDR.    Do I turn these off for callibration?

![image](https://github.com/user-attachments/assets/4acd1877-84c6-410d-b790-72561a6d56e4)

And this?
![image](https://github.com/user-attachments/assets/581a0ded-e8e1-4a30-9ec4-47372f56809a)

**Reply by awawa-dev** [reply]
Don't enable auto tone mapping for calibration. You can do it later if you need it so that HyperHDR automatically controls tone mapping but only after you have calibrated LUT (https://github.com/awawa-dev/HyperHDR/wiki/Automatic-tone-mapping). Tone mapping is irrelevant for calibration using MP4 video files, it will enable it by itself anyway. Quarter frame should be disabled for calibration, as in the instructions: https://github.com/awawa-dev/HyperHDR/wiki/lut-calibration

**Comment by Pavel2025-byte** [comment]
<img width="1027" alt="Снимок экрана 2024-12-18 в 23 35 09" src="https://github.com/user-attachments/assets/6f0b28df-3e5f-4455-9fa7-74446a636b05" />
Hi! I am using Ezcoo EZ-SP12H2 with MS2130 and Raspberry Pi Zero 2 W but I get wrong colors. Can you please tell me how to fix this?

**Reply by awawa-dev** [reply]
Maybe it's a Dolby Vision then only Low Latency Dolby Vision (LLDV) is supported by the splitter if the grabber is connected to Ezcoo output 2 and has scaler enabled for output 2. Or maybe this is a problem: https://github.com/awawa-dev/HyperHDR/discussions/848

**Reply by Pavel2025-byte** [reply]
Thanks, I'll try.

**Reply by Tchekevala** [reply]
I was getting a very bad color too, but i switch to another computer ( graphic card) everything look PERFECT now

**Comment by Muyagi** [comment]
Hi. I changed from MS2130 to UGREEN25173 and when I watch something from Netflix for example the grabber gets no video. With the MS2130 I hadn’t the problem. Rest of the hardware is the same. What could be the problem?

**Reply by awawa-dev** [reply]
It's hard to make a diagnosis or even a hypothesis from the description of the situation presented. UGreen 25173 supports much higher video bandwidth than MS2130 and it is an HDMI 2.0 grabber and not HDMI 1.4 like MS2130. I would bet on HDCP but since the rest hasn't changed and Ezcoo is still ahead of UGreen it should remove it...

**Reply by Muyagi** [reply]
I turned the scaler off for Out2, then it worked.
